src/Models/UserModel.php

```diff
@@ -208,4 +210,41 @@ private function checkQueryReturn(bool $return): void
             throw new RuntimeException($message, $error['code']);
         }
     }
+
+    /**
+     * Override the BaseModel's `save()` method to allow
+     * updating of user email, password, or password_hash
+     * fields if they've been modified.
+     *
+     * @param mixed $data
+     */
+    public function save($data): bool
+    {
+        try {
+            $result = parent::save($data);
+
+            if ($result && $data instanceof User) {
+                /** @var User $user */
+                $user = $data->id === null
```

Comment:
We use the method everywhere else, might as well be consistent. 
```suggestion
                $user = $data->getAuthId() === null
```

Reply:
But we're talking about getting rid the `getAuthX()` methods, so why use it more? But I'll change for now.

Reply:
Actually - turns out we can't use that because of the assert call. When creating a `new User()` and attempting to save it, there is no id, but the assertion yells anyway. Reverting.